waku/v2/node/waku_node.nim

```diff
@@ -361,21 +361,16 @@ proc publish*(node: WakuNode, topic: Topic, message: WakuMessage) {.async, gcsaf
   ## Publish a `WakuMessage` to a PubSub topic. `WakuMessage` should contain a
   ## `contentTopic` field for light node functionality. This field may be also
   ## be omitted.
-  ##
-  ## Status: Implemented.
     
-  if node.wakuRelay.isNil:
+  if node.wakuRelay.isNil():
```

Comment:
I don't think there is consensus yet, maybe a good addition to the nwaku style guide?

Reply:
invoking @jm-clius then, since he is writing a style guide :)

Reply:
Right, I don't have a strong preference here. Personally I find `echo x` to be the most readable, but for more obvious templates and procedures I like `isNil()` and `isOk()` with `()` (also because my editor highlighting works better this way).

I'm happy to add this (or whatever we have consensus on) to the coding/style guide.

However, Nim syntax inherently allows for some flexibility. In some cases where there is a clear benefit to preferring one alternative above the others, we should be prescriptive. Example here would be to avoid using `result` returns, avoid overly long lines, use `std/` prefix for standard imports, etc.
I'm worried though that being prescriptive _everywhere_ where there's flexibility, would tie contributors up in trying to follow a pedantic style guide rather than just understanding the language and its usage. An example here would be if we become prescriptive in using explicit `return` vs expression based returns. Not sure where the empty proc/template invocations above would fit in, but imo the impact of one choice above the other is low. The coding guide should be a simple reference, rather than keep us occupied with endless refactoring. Wdyt?

Reply:
yeap, totally agree. no need to be overly pedantic nor too strict in the coding guide.

imho agreeing that there is no consensus on `isOk/isOk()` is an actual consensus.

Perhaps add just a suggestion. "in these occasions we tend to use isOk and in these isOk() because of this". But no need to enforce it.

Reply:
I have an strong opinion on this. I am following these rules for the sake of consistency and readability:

- Verbs must be followed by the parenthesis, since it denotes a "question" (e.g., `isSomething()`, `isOk()`, `isNil()`)
- Nouns must go without the parenthesis. A `proc` with a noun name should go without parenthesis. They describe "properties" (or "derived properties" in the `proc` with a noun name) of certain object. (e.g., `Result.error`, `Result.value`)

I prefer languages more opinionated in this sense. Nim is a language prone to style inconsistency.

Reply:
@LNSD, fair point. I think this is a good general guideline for us to add.